scripts/Muon/GUI/Common/corrections_tab_widget/corrections_presenter.py

```diff
@@ -95,15 +100,68 @@ def handle_run_selector_changed(self) -> None:
         self.dead_time_presenter.handle_run_selector_changed()
         self.background_presenter.handle_run_selector_changed()
 
-    def handle_pre_process_and_grouping_complete(self) -> None:
-        """Handles when MuonPreProcess and grouping has been completed."""
-        self.dead_time_presenter.handle_pre_process_and_grouping_complete()
-        self.background_presenter.handle_pre_process_and_grouping_complete()
+    def handle_pre_process_and_counts_calculated(self) -> None:
+        """Handles when MuonPreProcess and counts workspaces have been calculated."""
+        self.dead_time_presenter.handle_pre_process_and_counts_calculated()
+        self.background_presenter.handle_pre_process_and_counts_calculated()
+
+    def handle_thread_calculation_started(self) -> None:
+        """Handles when a calculation on a thread has started."""
+        self.disable_editing_notifier.notify_subscribers()
+        self.thread_success = True
+
+    def handle_background_corrections_for_all_finished(self) -> None:
+        """Handle when the background corrections for all has finished."""
+        self.enable_editing_notifier.notify_subscribers()
+        if not self.thread_success:
+            return
+
+        self.background_presenter.handle_background_corrections_for_all_finished()
+
+        corrected_runs_and_groups = self.thread_model_wrapper.result
+        if corrected_runs_and_groups is not None:
+            self._perform_asymmetry_pairs_and_diffs_calculation(*corrected_runs_and_groups)
```

Comment:
why does this have a `*` in the brackets?

Reply:
It is used to unpack the tuple `corrected_runs_and_groups`. I will do this differently so that maybe its clearer

Reply:
Im only used to seeing it in C++ so was unsure what it was used for in Python.